File: javascript_bro/inheritance.js






class Animal{
    alive=true;
    eat(){
        console.log(`This ${this.name} is eating`);
    }
    sleep(){
        console.log(`This ${this.name} is sleeping`);
    }
}

class Rabbit extends Animal{
    name="rabbit";
    run(){
        console.log(`This ${this.name} is running`);
    }
}
class Fish extends Animal{
    name="fish";
    swim(){
        console.log(`This ${this.name} is swimming`);
    }
}

const RabbitObj=new Rabbit();
const FishObj=new Fish();
//console.log(RabbitObj.alive);
RabbitObj.eat();
RabbitObj.sleep();
RabbitObj.run();

FishObj.eat();
FishObj.swim();

File: python_bro/POOP/inheritance.py


class Animal:
    alive=True
    def eat(self):
        print("This animal is eating")
    def sleep(self):
        print("This animal is sleeeping")
        
class Rabbit(Animal):
    def run(self):
        print("This animal is running")

class Fish(Animal):
    pass

class Hawk(Animal):
    pass

rabit=Rabbit()
fish=Fish()
hawk=Hawk()

print(rabit.alive)
rabit.sleep()
rabit.run()